s3/src/main/resources/reference.conf

```diff
@@ -85,4 +85,9 @@ alpakka.s3 {
   # Which version of the list bucket api to use. Set to 1 to use the old style version 1 API.
   # By default the newer version 2 api is used.
   list-bucket-api-version = 2
+
+  # Object keys are validated to NOT use sub-directory selection with `..` to improve security.
+  # This flag may disable the validation.
+  # See https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingMetadata.html
+  validate-object-key = true
```

Comment:
Would it make sense to let the user toggle validating the bucket name too, for the same reason?

Reply:
Maybe the config could be generalized to apply validation to both the key and bucket name.

Reply:
That "feature" will not be available for DNS-based bucket names, so I think it is best to forbid that right away.